hexrdgui/llnl_import_tool_dialog.py

```diff
@@ -297,23 +409,149 @@ def update_bbox_width(self, val):
         self.it.scale_template(sx=scale)
 
     def _set_transform(self):
+        if self.instrument == 'FIDDLE':
+            flip = LLNLTransform.FIDDLE
         if self.instrument == 'PXRDIP':
             flip = LLNLTransform.PXRDIP
         elif self.instrument == 'TARDIS':
-            if self.detector == 'IMAGE-PLATE-2':
+            if self.image_plate == 'IMAGE-PLATE-2':
                 flip = LLNLTransform.IP2
-            elif self.detector == 'IMAGE-PLATE-3':
+            elif self.image_plate == 'IMAGE-PLATE-3':
                 flip = LLNLTransform.IP3
-            elif self.detector == 'IMAGE-PLATE-4':
+            elif self.image_plate == 'IMAGE-PLATE-4':
                 flip = LLNLTransform.IP4
         HexrdConfig().load_panel_state['trans'] = [flip]
 
+    def accept_detector(self, data_file, dark_file):
+        # Custom dark subtraction for the FIDDLE instrument
+        img = [[]] * FIDDLE_FRAMES
+        first, last = '/'.join(FIDDLE_HDF5_PATH).rsplit('0', 1)
+        for frame in range(FIDDLE_FRAMES):
+            # Use known path to find data 0/1/2/3 paths
+            path = f'{first}{frame}{last}'
+            # Load in detector data and dark data for this frame
+            with h5py.File(data_file) as data:
+                data_raw = np.array(data[path])
+            with h5py.File(dark_file) as dark:
+                dark_raw = np.array(dark[path])
```

Comment:
```suggestion
            with h5py.File(dark_file) as dark:
                dark_raw = np.array(dark[path])
```

Looks like these lines were duplicated

Reply:
Actually one is for the data file, the other for the dark file

Reply:
Ah I see! :slightly_smiling_face: